bc_obps/service/facility_service.py

```diff
@@ -16,9 +16,120 @@
+        2. **Include Specific Fields:** The method specifically includes fields such as 'name', 'type', 'latitude_of_largest_emissions', and 'longitude_of_largest_emissions'.
+
+        Parameters:
+        - `payload` (FacilityIn): The data payload containing facility details that need to be processed.
+
+        Returns:
+        - `dict`: A dictionary containing the extracted facility data.
+        """
+        # Prepare a dictionary of facility data by including specified fields from the payload
+        return payload.dict(
+            include={
+                'name',
+                'type',
+                'latitude_of_largest_emissions',
+                'longitude_of_largest_emissions',
+            }
+        )
+
+    @classmethod
+    def create_or_update_address(cls, payload: FacilityIn) -> Optional[Address]:
+        """Helper function to create or update an address from payload data."""
+        address_data = payload.dict(
+            include={'street_address', 'municipality', 'province', 'postal_code'}, exclude_none=True
+        )
+        if address_data:
+            return AddressDataAccessService.create_address(address_data)
+        return None
+
+    @classmethod
+    def handle_well_authorization_numbers(cls, user_guid: UUID, payload: FacilityIn, facility: Facility) -> None:
+        """
+        Helper function to process and set well authorization numbers for a facility.
+
+        This method handles the addition and removal of well authorization numbers for a given facility based on
+        the provided payload. It ensures that there are no duplicate well authorization numbers in the new set
+        and updates the facility's well authorization numbers accordingly.
+
+        Args:
+            user_guid (UUID): The GUID of the user making the changes.
+            payload (FacilityIn): The payload containing the new well authorization numbers.
+            facility (Facility): The facility object to be updated.
+
+        Raises:
+            Exception: If there are duplicate well authorization numbers in the new set.
+
+        Steps:
+            1. Extract existing well authorization numbers from the facility.
+            2. Compare new well authorization numbers from the payload with the existing ones.
+            3. Identify and add new well authorization numbers that are not in the existing set.
+            4. Identify and remove old well authorization numbers that are not in the new set.
+        """
+        existing_numbers = set(facility.well_authorization_numbers.values_list('well_authorization_number', flat=True))
+        new_numbers = payload.well_authorization_numbers
+
+        # Check for duplicates within the new_numbers
+        if len(new_numbers) != len(set(new_numbers)):
+            raise Exception(
+                "Well Authorization Number: Well authorization number with this Well authorization number already exists."
+            )
```

Comment:
We could also do this with a db constraint for uniqueness, which is how we've done it for facility name and other fields. @Sepehr-Sobhani , do you know if there's a preferred approach?

Reply:
In order to enforce this constraint, we will need to create a [through table](https://docs.djangoproject.com/en/5.0/topics/db/models/#extra-fields-on-many-to-many-relationships) and add a UniqueConstraint to that table. This will involve more changes than the current PR. I think the code to check for duplicate values should suffice for now. However, I recommend using a different error message to indicate that the user has entered repeated numbers.

Reply:
@BCerki  @Sepehr-Sobhani 

The message `Well Authorization Number: Well authorization number with this Well authorization number already exists.` mimics the message return to the user when trying to add a number that exists in another record. From this method, I believe:
```
bc_obps/service/data_access_service/well_authorization_number_service.py/ def create_well_authorization_number
 WellAuthorizationNumber.objects.create(
            well_authorization_number=number,
            created_by_id=user_guid,
        )
```


![image](https://github.com/user-attachments/assets/648d14d7-65fe-4116-b1c8-22c7c15b933b)

Reply:
But it seems like we are checking new incoming numbers that are not duplicates, instead of the existing ones in the database, am I correct? 🤔 Ignore my suggestion if I'm wrong lol

Reply:
@Sepehr-Sobhani 
You are correct :) Thank-you

Reply:
I didn't see a check against the existing numbers, did I miss it?